slatedb/src/merge_operator.rs

```diff
@@ -63,10 +66,37 @@ pub trait MergeOperator {
         existing_value: Option<Bytes>,
         value: Bytes,
     ) -> Result<Bytes, MergeOperatorError>;
+
+    /// Merges a batch of operands with an optional existing value.
+    ///
+    /// This method allows for more efficient batch processing of merge operands.
+    /// The default implementation applies pairwise merging, but implementations
+    /// can override this for better performance (e.g., a counter can sum all values at once).
+    ///
+    /// # Arguments
+    /// * `existing_value` - The current accumulated value (if any)
+    /// * `operands` - A slice of operands to merge, ordered from oldest to newest
+    ///
+    /// # Returns
+    /// * `Ok(Bytes)` - The merged result as bytes
+    /// * `Err(MergeOperatorError)` - If the merge operation fails
+    fn merge_batch(
```

Comment:
this API LGTM but can you add it to https://github.com/slatedb/slatedb/blob/main/rfcs/0006-merge-operator.md to document the public API change? we will also need two +1s to change the public API